include/circt/Dialect/FIRRTL/FIRAnnotations.h

```diff
@@ -23,13 +25,20 @@ class Value;
 } // namespace json
 } // namespace llvm
 
-namespace mlir {
-class ArrayAttr;
-}
```

Comment:
@lattner had pushed me in the direction of more forward declarations and fewer includes. There may be a way to do the same thing here with the `DictionaryAttr` and removing the added includes above.

Reply:
I'm not totally convinced that my annotation helpers are generically useful enough, maybe I should just remove them?

Reply:
Since they're currently single-use, maybe remove them and we can grow the utilities API as needed. Good idea.